src/perl/Permabit/BlockDevice/VDO/LVMVDO/ThinPool.pm

```diff
@@ -0,0 +1,176 @@
+##
+# Perl object that represents an LVM Thin Pool using a VDO as data device
+#
+# $Id$
+##
+package Permabit::BlockDevice::VDO::LVMVDO::ThinPool;
+
+use strict;
+use warnings FATAL => qw(all);
+use Carp qw(confess);
+use English qw(-no_match_vars);
+use Log::Log4perl;
+use Permabit::Assertions qw(
+  assertDefined
+  assertNumArgs
+  assertType
+);
+use Permabit::Constants;
+use Permabit::ProcessUtils qw(delayFailures);
+use Permabit::Utils qw(makeFullPath);
+
+use base qw(
+  Permabit::BlockDevice::VDO::LVMVDO
+  Permabit::BlockDevice::LVM::ThinPool
+);
+
+my $log = Log::Log4perl->get_logger(__PACKAGE__);
+
+########################################################################
+# @inherit
+##
+sub setup {
+  my ($self) = assertNumArgs(1, @_);
+
+  # Get the location of the vdoformat binary for use by LVM.
+  $self->{vdoformat} = $self->getMachine()->findNamedExecutable("vdoformat");
+
+  # This will create a thin pool that has vdo set up to be the data device
+  my $config = $self->makeLVMConfigString();
+  my $totalSize = $self->getTotalSize($self->{physicalSize});
+  $self->{volumeGroup}->createThinPool($self->{deviceName},
+                                       $totalSize,
+                                       $config,
+                                       1);
+  $self->Permabit::BlockDevice::VDO::setup();
+}
+
+########################################################################
+# @inherit
+##
+sub growLogical {
+  my ($self, $logicalSize) = assertNumArgs(2, @_);
+
+  # resize the vdo logical volume
+  my $name = "$self->{deviceName}";
+
+  my $newSize = $self->{volumeGroup}->alignToExtent($logicalSize);
+  $self->{volumeGroup}->resizeLogicalVolume($name, $newSize);
+  $self->{logicalSize} = $newSize;
+}
+
```

Comment:
growPhysical() and growLogical() both look like they have the same implementation in both subclasses of LVMVDO. If this is actaully treu (and not just me misreading), then they should stay in the superclass.

Reply:
growphysical and growlogical have been stubbed out for now for thinpoolvdo. Managed still grows the devices as before.